src/v0.8/module/dataset/DatasetsProof.sol

```diff
@@ -204,8 +232,23 @@ contract DatasetsProof is
                 ),
                 "Invalid mappingFiles percentage"
             );
-            datasets.reportDatasetProofSubmitted(_datasetId);
-            emit DatasetsEvents.DatasetProofSubmitted(_datasetId, msg.sender);
+
+            if (
+                escrow.getOwnerCollateral(
+                    EscrowType.Type.DatacapCollateral,
+                    datasets.getDatasetMetadataSubmitter(_datasetId),
+                    _datasetId
+                ) < getDatasetCollateral(_datasetId)
```

Comment:
Both SC collateral and DA fee payment collateral need to be processed simultaneously.

Reply:
This issue only include escrow and DatacapCollateral logic